### ISSUE TITLE
[sda-download] Get public key for receiver reencryption

### ISSUE THREAD

**Issue by pontus** [neicnordic/sensitive-data-archive#364]
sda-download currently does not support encrypted downloads, but it should (see e.g. #361) 

To be able to reencrypt the requested data for the user, it needs the public key of the receiver. If it can't get it, it should fail the request.

Generally I think we should support accepting the receiver public key in the request, but to support compatibility with "normal" s3 clients, we should also support having a per-user default key (settable somehow through a web interface, out of scope for this issue but the interface needs to be defined).

**Comment by pontus** [comment]
Makes sense to me to leave that to a separate microservice, but assuming we don't opt for required pre-registration of keys (which I imagine would exclude some desired use cases), `sda-download` still needs to be able to accept a key somehow and that interface should be defined.

**Comment by blankdots** [comment]
i am fine taking a stab at it with a proposal, so I will assign myself for this, just to signal that I am working with an idea on it.
For now I will work in `sda-download` repo and my idea would be based on grpc what i mentioned above

**Comment by blankdots** [comment]
relates to #667

**Comment by blankdots** [comment]
relates to #698 ?
